text_handler.py

```diff
@@ -42,13 +42,15 @@ async def __is_provide_service(self, message: types.Message):
             return True
         chat_type = message.chat.type
         if chat_type == "private":
-            member = await self.bot.get_chat_member(GROUP_CHAT_ID, chat_id)
-            if member.is_chat_member():
-                return True
-            else:
-                await message.reply(AVAILABILITY_HTML,
-                                    parse_mode=ParseMode.HTML)
-                return False
+            user_id = chat_id
+            for promotion_chat in PROMOTION_LIST_CHAT_ID:
+                if (await self.bot.get_chat_member(
+                        promotion_chat, user_id)).is_chat_member():
+                    return True
+                else:
+                    await message.reply(AVAILABILITY_HTML,
+                                        parse_mode=ParseMode.HTML)
+                    return False
```

Comment:
i mean why would we need to add the bot in the channel anyway?

Reply:
This probably also applies to the group, if the PROMOTION_LIST_CHAT_ID has a group where the bot is not a member, it will also fails to get members list and output errors (won't terminate the bot, it will keep running fine but just won't send AVAILABILITY_HTML message)

In your case, you probably don't need to add a channel id to PROMOTION_LIST_CHAT, but only the group id one or more ones.

iirc telegram channels and groups are basically same for telegram servers, they just have different permissions. in channels that only admin can see list of members and members can't send messages. in group, every member could see member list and can send message by default.

in my case, when I need to share the bot with people, it's not necessarily to join the group, but join a channel instead (it's also good that they won't see each other in member list). it's also a good way to promote the channel.